sn_transfers/src/cashnotes/signed_spend.rs

```diff
@@ -152,64 +84,42 @@ impl SignedSpend {
 
-        // Check that the parent where all spent to our parent_tx
-        let tx_our_cash_note_was_created_in = self.parent_tx_hash();
-        let mut actual_parent_spends = BTreeSet::new();
-        for parents in parent_spends.clone().into_iter() {
-            if parents.is_empty() {
-                error!("No parent spend provided for {unique_key}");
-                return Err(TransferError::InvalidParentSpend(
-                    "Parent is empty".to_string(),
-                ));
-            }
-            let parent_unique_key = parents
-                .iter()
-                .map(|p| *p.unique_pubkey())
-                .collect::<BTreeSet<_>>();
-            if parent_unique_key.len() > 1 {
+        let mut total_inputs: u64 = 0;
+        for p in parent_spends {
+            if p.len() > 1 {
                 error!("While verifying parents of {unique_key}, found a parent double spend, but it contained more than one unique_pubkey. This is invalid. Erroring out.");
                 return Err(TransferError::InvalidParentSpend("Invalid parent double spend. More than one unique_pubkey in the parent double spend.".to_string()));
             }
 
-            // if parent is a double spend, get the actual parent among the parent double spends
-            let actual_parent = parents
-                .iter()
-                .find(|p| p.spent_tx_hash() == tx_our_cash_note_was_created_in)
-                .cloned();
-
-            match actual_parent {
-                Some(actual_parent) => {
-                    actual_parent_spends.insert(actual_parent);
-                }
-                None => {
-                    let tx_parent_was_spent_in = parents
-                        .iter()
-                        .map(|p| p.spent_tx_hash())
-                        .collect::<Vec<_>>();
+            if let Some(parent) = p.first() {
+                if let Some(amount) = parent.spend.get_output_amount(unique_key) {
+                    total_inputs += amount.as_nano();
+                } else {
                     return Err(TransferError::InvalidParentSpend(format!(
-                        "Parent spend was spent in another transaction. Expected: {tx_our_cash_note_was_created_in:?} Got: {tx_parent_was_spent_in:?}"
+                        "Parent spend {:?} doesn't contain self spend {unique_key:?} as one of its output", parent.unique_pubkey() 
                     )));
                 }
+            } else {
+                warn!("While verifying parents of {unique_key}, found a parent with empty spend");
+                return Err(TransferError::InvalidParentSpend(
+                    "Invalid parent empty spend.".to_string(),
+                ));
             }
         }
 
-        // Here we check that the CashNote we're trying to spend was created in a valid tx
-        if let Err(e) = self
-            .spend
-            .parent_tx
-            .verify_against_inputs_spent(actual_parent_spends.iter())
-        {
```

Comment:
We should add back these checks:
- we are an output of parent (same pk same value)
- parent is an input of us (same pk same value)